EIPS/eip-1895.md

```diff
@@ -0,0 +1,158 @@
+
+```.
+p = 475922286169261325753349249653048451545124879242694725395555128576210262817955800483758081
+```
+
+with generator P:
+
+```.
+P = (
+    60760244141852568949126569781626075788424196370144486719385562369396875346601926534016838,
+    363732850702582978263902770815145784459747722357071843971107674179038674942891694705904306
+)
+```
+
+Both p and q can be written in 298 bits.
+
+The group G_1 is defined on the curve defined by the equation `Y² = X³ + aX + b` where:
+
+```.
+    a = 2
+    b = 423894536526684178289416011533888240029318103673896002803341544124054745019340795360841685
+```
+
+The twisted group G_2 is defined over the field `F_p^2 = F_p / <<To be completed>>`
+
+The twisted group G_2 is defined on the curve defined by the equation `Y² = X² + aX + b` where :
+
+```.
+    a = 34 + i * 0
+    b = 0 + i * 67372828414711144619833451280373307321534573815811166723479321465776723059456513877937430
+```
+
+G_2 generator is generated by :
+
+```.
+    P2 = (
+        438374926219350099854919100077809681842783509163790991847867546339851681564223481322252708 +
+        i * 37620953615500480110935514360923278605464476459712393277679280819942849043649216370485641,
+        37437409008528968268352521034936931842973546441370663118543015118291998305624025037512482 +
+        i * 424621479598893882672393190337420680597584695892317197646113820787463109735345923009077489
+    )
+```
+
+### The operations and gas cost
+
+The following operations and their gas cost would be implemented
+
+```.
+MNT_X_ADD = <<To be estimated>>
+MNT_X_MUL = <<To be estimated>>
+MNT_X_PAIRING = <<To be estimated>>
+```
+
+Where `X` is either 4.
+
+### Encoding
+
+The curves points P(X, Y) over F_p are represented in their compressed form C(X, Y):
+
+```.
+    C = X | s
+```
+
+where `s` represents `Y` as follow:
+
+```.
+    |  `s'`  | `Y`                      |
+    |--------|--------------------------|
+    | `0x00` | Point at infinity        |
+    | `0x02` | Solution with `y` even   |
+    | `0x03` | Solution with `y` odd    |
+```
+
+Compression operation from affine coordinate is trivial:
+
+```.
+    s = 0x02 | (s & 0x01)
+```
+
+In the EVM? the compressed form allows us to represents curve points with 2 uint256 instead of 3.
+
+### Edge cases
+
+* Several acceptable representations for the point at infinity
+
+## Rationale
+
+The only usable cycle in my knowledge are introduced in this [paper](https://eprint.iacr.org/2014/595.pdf).
+
+It has 80 bits (the curves respectively have 80 and 120 bits of security) of security which might be considered not enough for some application, but enough for others.
+
+If not, another curve is being used by Coda but is defined over a 727 bits which might be considered prohibitively slow. *Note, that the curves and field elements are represented on field larger than 256 bits (even for the 80 bits of security), therefore it might be necessary to add support for larger field size operations.
```

Comment:
Why add your rationale as a note? It seems like a perfectly valid argument that could just be a normal sentence as the others.